File: algorithms/2269.find-the-k-beauty-of-a-number/solution.rb
# @param {Integer} num
# @param {Integer} k
# @return {Integer}
def divisor_substrings(num, k)
  count = 0

  num.to_s.chars.each_cons(k) do |v|
    i = v.join.to_i
    next if i == 0
    count += 1 if num % i == 0
  end

  count
end

File: algorithms/2269.find-the-k-beauty-of-a-number/solution.rs
impl Solution {
    pub fn divisor_substrings(num: i32, k: i32) -> i32 {
        let digits: Vec<char> = num.to_string().chars().collect();

        let k_usize = k as usize;
        let mut count = 0i32;
        for i in 0..=digits.len() - k_usize {
            let s: String = (&digits[i..i + k_usize]).iter().collect();
            match s.parse::<i32>() {
                Ok(i) => {
                    if i != 0 && num % i == 0 {
                        count += 1;
                    }
                }
                Err(e) => {
                    unreachable!()
                }
            }
        }

        count
    }
}
